ansi/parser.go

```diff
@@ -228,10 +232,16 @@ func (p *Parser) performAction(dispatcher ParserDispatcher, action parser.Action
 		p.Cmd |= int(b) << parser.MarkerShift
 
 	case parser.CollectAction:
-		// Collect intermediate bytes
-		// we only store the last intermediate byte
-		p.Cmd &^= 0xff << parser.IntermedShift
-		p.Cmd |= int(b) << parser.IntermedShift
+		if state == parser.Utf8State {
+			p.collectRune(b)
+		} else {
+			// Collect intermediate bytes
+			// we only store the last intermediate byte
+			p.Cmd &^= 0xff << parser.IntermedShift
+			// p.Cmd[2] &^= 0xff
+			p.Cmd |= int(b) << parser.IntermedShift
+			// p.Cmd[2] |= b
```

Comment:
I'm unsure if these commented lines are left behind debugs used for dev, or a comment

```suggestion
			p.Cmd &^= 0xff << parser.IntermedShift
			p.Cmd |= int(b) << parser.IntermedShift
```

Reply:
Whops, how did that get in there 😄

Reply:
Updated the commit, thanks!